### PR TITLE
Further eRS APIM integration / end to end tests

### PR DESCRIPTION
**Pull request by andrewclark2020**
## Summary
* :sparkles: New Feature
New tests added which hit the actual eRS endpoint and verify the headers received

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

**Review by acarriedev — DISMISSED** [review]
Looks for from a mock auth perspective

**Review by nhsd-david-wass — APPROVED** [review]
Looks all good to me

File: tests/integration/test_headers.py [hunk 1]
```diff
@@ -1,88 +1,216 @@
...
+        response = requests.get(
+            f"{service_url}{SPECIALTY_REF_DATA_URL}", headers=client_request_headers
+        )
+
+        # Verify the status
+        assert (
+            response.status_code == 401
+        ), "Expected a 401 when accesing the api but got " + (str)(response.status_code)
+
+        # Verify the response headers
```
Comment (andrewclark2020) [inline tests/integration/test_headers.py:202]: Assert no body on oauth error

File: tests/integration/test_headers.py [hunk 2]
```diff
@@ -1,88 +1,216 @@
...
+        return self.value
+
+
+class RenamedHeader(Enum):
+    CORRELATION_ID = "nhsd-correlation-id"
+    BUSINESS_FUNCTION = "x-ers-business-function"
+    ODS_CODE = "x-ers-ods-code"
+    FILENAME = "x-ers-xapi-meta-file_name"
+    COMM_RULE_ORG = "x-ers-xapi-comm-rule-org"
+    REFERRAL_ID = "x-ers-xapi-meta-intended_ubrn"
```
Comment (andrewclark2020) [inline tests/integration/test_headers.py:43]: Can these be combined nicely?

**Review by nhsd-nigel-mellor — APPROVED** [review]

**Pull request merged**